modules/flyte/iam.tf

```diff
@@ -13,7 +13,7 @@ resource "aws_iam_role" "flyte_controlplane" {
           StringEquals : {
             "${trimprefix(local.oidc_provider_url, "https://")}:aud" : "sts.amazonaws.com",
             "${trimprefix(local.oidc_provider_url, "https://")}:sub" : [
-              "system:serviceaccount:${var.platform_namespace}:${var.serviceaccount_names.flyteadmin}",
+              "system:serviceaccount:${var.platform_namespace}:${var.serviceaccount_names.datacatalog}",
```

Comment:
Leaving datacatalog with controlplane role for now until we confirm that it needs data bucket access

Reply:
I think you're right here; input file uploads are just wonky ([issue](https://github.com/flyteorg/flyte/issues/5051))